src/hb-ot-layout.cc

```diff
@@ -1989,6 +1989,116 @@ hb_ot_layout_get_baseline (hb_font_t                   *font,
 
   return result;
 }
+
+hb_bool_t
+hb_ot_layout_get_baseline_with_fallback (hb_font_t                   *font,
+					 hb_ot_layout_baseline_tag_t  baseline_tag,
+					 hb_direction_t               direction,
+					 hb_tag_t                     script_tag,
+					 hb_tag_t                     language_tag,
+					 hb_position_t               *coord        /* OUT.  May be NULL. */)
+{
+  if (coord)
+    *coord = 0;
+
+  if (hb_ot_layout_get_baseline (font,
+                                 baseline_tag,
+                                 direction,
+                                 script_tag,
+                                 language_tag,
+                                 coord))
+    return true;
+
+  if (!coord)
+    return false;
+
+  /* Synthesize missing baselines.
+   * See https://www.w3.org/TR/css-inline-3/#baseline-synthesis-fonts
+   */
+  switch ((int) baseline_tag)
+  {
+  case HB_OT_LAYOUT_BASELINE_TAG_IDEO_FACE_BOTTOM_OR_LEFT:
+  case HB_OT_LAYOUT_BASELINE_TAG_IDEO_FACE_TOP_OR_RIGHT:
+    {
+      hb_codepoint_t glyph;
+      hb_glyph_extents_t extents;
+
+      if (hb_font_get_nominal_glyph (font, 0x638, &glyph) &&
+          hb_font_get_glyph_extents (font, glyph, &extents))
+      {
+        *coord = extents.y_bearing;
+        if (baseline_tag == HB_OT_LAYOUT_BASELINE_TAG_IDEO_FACE_BOTTOM_OR_LEFT)
+          *coord += extents.height;
```

Comment:
Should be height or width depending on direction.

Reply:
Err. Ignore me. Humm.

Reply:
0x638 doesn't look right.

Reply:
I've removed the ideo-face baselines from this for now.